kbase-extension/static/kbase/js/common/jobManager.js

```diff
@@ -64,212 +120,123 @@ define(['common/html', 'common/jobs', 'common/ui', 'util/string'], (html, Jobs,
-         */
-        executeActionByJobStatus(args) {
-            const { action, statusList, validStatuses } = args;
-
-            const jobsList = this.getJobIDsByStatus(statusList, validStatuses);
-            if (!jobsList || !jobsList.length) {
-                return Promise.resolve(false);
+            // add the handler -- the message type is used as the event
+            if (handlerObject) {
+                this.addHandler(type, handlerObject);
             }
-
-            return UI.showConfirmDialog(
-                this._generateDialogArgs(action, statusList, jobsList)
-            ).then((confirmed) => {
-                if (!confirmed) {
-                    return false;
-                }
-                this.bus.emit(jobCommand[action], {
-                    jobIdList: jobsList,
-                });
-                return true;
-            });
         }
 
         /**
-         * Cancel a single job from the batch
+         * Remove the listener for ${type} messages for a job ID
          *
          * @param {string} jobId
+         * @param {string} type - the type of the listener
          */
-        cancelJob(jobId) {
-            return this.executeActionOnJobId({ jobId, action: 'cancel' });
+        removeListener(jobId, type) {
+            try {
+                this.bus.removeListener(this.listeners[jobId][type]);
+                delete this.listeners[jobId][type];
+            } catch (err) {
+                // do nothing
+            }
         }
 
         /**
-         * Retry a single job from the batch
-         *
+         * Remove all listeners associated with a certain job ID
          * @param {string} jobId
          */
-        retryJob(jobId) {
-            return this.executeActionOnJobId({ jobId, action: 'retry' });
-        }
-
-        /**
-         * Cancel all jobs with the specified statuses
-         *
-         * @param {array} statusList - array of statuses to cancel
-         */
-        cancelJobsByStatus(statusList) {
-            return this.executeActionByJobStatus({
-                action: 'cancel',
-                statusList: statusList,
-                validStatuses: Jobs.validStatusesForAction.cancel,
-            });
-        }
-
-        /**
-         * Retry all jobs that ended with the specified status(es)
-         *
-         * @param {array} statusList - array of statuses to retry
-         */
-        retryJobsByStatus(statusList) {
-            return this.executeActionByJobStatus({
-                action: 'retry',
-                statusList: statusList,
-                validStatuses: Jobs.validStatusesForAction.retry,
-            });
+        removeJobListeners(jobId) {
+            if (this.listeners[jobId] && Object.keys(this.listeners[jobId]).length) {
+                Object.keys(this.listeners[jobId]).forEach((type) => {
+                    this.bus.removeListener(this.listeners[jobId][type]);
+                    delete this.listeners[jobId][type];
```

Comment:
No `try/catch` here as above? You could also just call `removeListener` on each `jobId` and `type`.

Reply:
Just realised that - doh! Thanks for the prompt.